agr_java_core/src/main/java/org/alliancegenome/neo4j/view/ParalogBean.java

```diff
@@ -0,0 +1,67 @@
+package org.alliancegenome.neo4j.view;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.annotation.JsonView;
+import lombok.Data;
+import org.alliancegenome.neo4j.entity.node.Gene;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+import java.io.Serializable;
+import java.util.List;
+
+@Data
+@Schema(name="OrthologView", description="POJO that represents the Paralog view")
+@JsonPropertyOrder({"gene", "homologGene", "length", "similarity", "identity", "rank", "predictionMethodsMatched", "predictionMethodsNotMatched", "predictionMethodsNotCalled", "methodCount", "totalMethodCount"})
+public class ParalogBean implements Serializable {
+
+	@JsonView(View.Homology.class)
+	private Gene gene;
+	@JsonView(View.Homology.class)
+	private Gene homologGene;
+
+	@JsonView(View.Homology.class)
+	private Float length;
+	@JsonView(View.Homology.class)
+	private Float similarity;
+	@JsonView(View.Homology.class)
+	private Integer rank;
+	@JsonView(View.Homology.class)
+	private Float identity;
+
+	@JsonView(View.Homology.class)
+	private List<String> predictionMethodsNotCalled;
+	@JsonView(View.Homology.class)
+	private List<String> predictionMethodsMatched;
+	@JsonView(View.Homology.class)
+	private List<String> predictionMethodsNotMatched;
+
+	@JsonView(View.Homology.class)
+	@JsonProperty(value = "methodCount")
+	public Integer getMethodCount() {
+		if (predictionMethodsMatched == null)
+			return 0;
+		return predictionMethodsMatched.size();
+	}
+
+	@JsonProperty(value = "methodCount")
+	public void setMethodCount(Integer count) {
+	}
+
+	@JsonView(View.Homology.class)
+	@JsonProperty(value = "totalMethodCount")
+	public Integer getTotalMethodCount() {
+		if (predictionMethodsMatched == null && predictionMethodsNotMatched == null)
+			return 0;
+		if (predictionMethodsMatched == null)
+			return predictionMethodsNotMatched.size();
```

Comment:
If predictionMethodsNotMatched is null won't this through a NPE?

Reply:
Is one of these always non null?

Reply:
I think so but if both can be null this code handles this as well.

Reply:
The code is using a AND which means that one of them could be null.

Reply:
If they're both null then the code returns 0.  If you get past that then if predictionMethodsMatched is null then predictionMethodsNotMatched cannot be null (otherwise you would have already returned 0).